node/src/actors/chain_manager/mod.rs

```diff
@@ -1085,15 +1103,20 @@ impl ChainManager {
         );
         // Ignore AddTransaction when not in Synced state
         match self.sm_state {
-            StateMachine::Synced => {}
-            _ => {
-                return Box::pin(actix::fut::err(
-                    ChainManagerError::NotSynced {
-                        current_state: self.sm_state,
-                    }
-                    .into(),
-                ));
-            }
+            StateMachine::Synced | StateMachine::AlmostSynced => {}
```

Comment:
This change looks important. Maybe this explains why some blocks have zero transactions?

Reply:
Maybe, it something that we didnt modify well when we include the AlmostSynced state